library/src/main/java/com/bumptech/glide/RegistryFactory.java

```diff
@@ -279,11 +279,15 @@ Uri.class, Bitmap.class, new ResourceBitmapDecoder(resourceDrawableDecoder, bitm
           DirectResourceLoader.inputStreamFactory(context);
       ModelLoaderFactory<Integer, AssetFileDescriptor> assetFileDescriptorFactory =
           DirectResourceLoader.assetFileDescriptorFactory(context);
+      ModelLoaderFactory<Integer, Drawable> drawableFactory =
+          DirectResourceLoader.drawableFactory(context);
       registry
           .append(int.class, InputStream.class, inputStreamFactory)
           .append(Integer.class, InputStream.class, inputStreamFactory)
           .append(int.class, AssetFileDescriptor.class, assetFileDescriptorFactory)
           .append(Integer.class, AssetFileDescriptor.class, assetFileDescriptorFactory)
+          .append(int.class, Drawable.class, drawableFactory)
+          .append(Integer.class, Drawable.class, drawableFactory)
```

Comment:
**Suggestion**: Validate the integration of the new drawableFactory with the existing registry setup. Ensure that it does not introduce conflicts with other registered loaders, especially in scenarios involving resource and drawable loading. <br> **Code Suggestion**: 
 ```
 + // Ensure compatibility and no conflicts with existing loaders
 ```